addons/mail/static/src/js/activity.js

```diff
@@ -682,6 +686,14 @@ var KanbanActivity = BasicActivity.extend({
      */
     _render: function () {
         var $span = this.$('.o_activity_btn > span');
+        // when no goupby on kanban, it will not call renderElement so update activity icon manually.
+        if (this.recordData.activity_exception_decoration) {
+            $span.removeClass('fa-clock-o');
+            $span.addClass('text-' + this.recordData.activity_exception_decoration + ' ' + this.recordData.activity_exception_icon);
+        } else {
+            $span.removeClass();
+            $span.addClass('fa-lg fa-fw fa fa-clock-o');
```

Comment:
Mmmmh I don't like this ; if you change a class in the template, you also have to repeat the change here, which is not ideal.

Instead, I would use two different `<span>` in the template (you can use a special class to identify them) and toggle them according to `activity_exception_decoration`.

Reply:
@mge-odoo Here in each icon is different base on an activity type and we need to update icon accordingly so we haven't any particular class to remove.

Reply:
@mpa-odoo the special class I mentioned was just meant to be able to identify and differentiate the two `<span>` in the template, like I mentioned (like `<span class='o_activity_with_exception ...'/>` and `<span class='o_activity_without_exception fa fa-clock-o fa-lg fa-fw ...'/>`) ; do you understand why I want to `<span>`?

Reply:
@mge-odoo  yeah , got it. Thank you